src/Data/Fin/Base.agda

```diff
@@ -77,11 +77,24 @@ fromℕ<″ zero    (ℕ.less-than-or-equal refl) = zero
 fromℕ<″ (suc m) (ℕ.less-than-or-equal refl) =
   suc (fromℕ<″ m (ℕ.less-than-or-equal refl))
 
--- raise m "i" = "m + i".
+-- canonical liftings of smaller i:Fin m to larger index Fin (n + m)
 
+-- injection on the left: "i" ↑ˡ n = "i"
+_↑ˡ_ : ∀ {m} → Fin m → ∀ n → Fin (m ℕ.+ n)
+zero    ↑ˡ n = zero
+(suc i) ↑ˡ n = suc (i ↑ˡ n)
+
+-- injection on the right: n ↑ʳ "i" = "n + i".
+_↑ʳ_ : ∀ {m} n → Fin m → Fin (n ℕ.+ m)
+zero    ↑ʳ i = i
+(suc n) ↑ʳ i = suc (n ↑ʳ i)
+
+-- raise n "i" = "n + i".
 raise : ∀ {m} n → Fin m → Fin (n ℕ.+ m)
```

Comment:
If you want to keep the type, then could you put it next to the definition in the deprecated section? Although usually we just ditch the type entirely!

Reply:
Ah. I wasn't sure what the preferred workflow here was. But yes, if you're serious about the deprecation, then for sure make it harder to maintain the old, deprecated usage(s)... I get it ;-)